core/utils/file_utils.cpp

```diff
@@ -872,8 +854,8 @@ handle_t open(void* file, OpenMode mode, int advice) noexcept {
   // under /proc/self/fd the link is guaranteed to point to the original inode
   // even if the original file was removed
   auto fd = handle_cast(file);
-  char path[strlen("/proc/self/fd/") + sizeof(fd) * 3 +
-            1];  // approximate maximum number of chars, +1 for \0
+  // approximate maximum number of chars, +1 for \0
```

Comment:
NIT
we can make `/proc/self/fd/%d` a `constexpr string_view` and use `size()-2` instead of magic number "14"and `data()` below

Reply:
Will be done in the next PR